src/components/TodoModal/TodoModal.tsx

```
@@ -1,42 +1,79 @@
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+import { actions as currentTodoActions } from '../../features/currentTodo';
 
 export const TodoModal: React.FC = () => {
+  const [loader, setLoader] = useState(true);
+  const [user, setUser] = useState<User | null>(null);
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const dispatch = useAppDispatch();
+
+  const closeModal = () => dispatch(currentTodoActions.removeTodo());
+
+  const loadUser = useCallback(async () => {
+    try {
+      if (currentTodo) {
+        const fetchUser = await getUser(currentTodo.userId);
+
+        setUser(fetchUser);
+      }
+
+      setLoader(false);
+    } catch (error) {
+      // console.log(error);
```

don't leave empty catch block